File: backend/db/mongodb.js
import { MongoClient } from "mongodb";
import dotenv from "dotenv";
dotenv.config();

let db;
let client;

export async function connectDB() {
    try {
        client = new MongoClient(process.env.MONGODB_URI);
        await client.connect();
        db = client.db("ai_tutor");
        console.log("✅ Connected to MongoDB Atlas");
    } catch (error) {
        console.error("❌ MongoDB connection error:", error);
        process.exit(1);
    }
}

export function getDB() {
    if (!db) throw new Error("Database not initialized. Call connectDB() first.");
    return db;
}

File: python-service/db/mongodb.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

load_dotenv()

_client = None
_collection = None


def get_collection():
    global _client, _collection
    if _collection is None:
        _client = MongoClient(os.getenv("MONGODB_URI"))
        db = _client["ai_tutor"]
        _collection = db["placement_docs"]
        print("✅ Connected to MongoDB Atlas")
    return _collection
